modules/lang-painless/src/main/java/org/elasticsearch/painless/AnalyzerCaster.java

```diff
@@ -30,6 +30,9 @@
 public final class AnalyzerCaster {
 
     public static Cast getLegalCast(Location location, Type actual, Type expected, boolean explicit, boolean internal) {
+        if (actual == null || expected == null) {
```

Comment:
I added this because I hit an NPE deep in here and I figured I'd explicitly check it so we could give a nicer warning. Seeing this bug while compiling is always a bug on our part but at least we'll have more information than "NPE".